compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/AMDAIETileSizeSelectionUtilsTest.cpp

```diff
@@ -67,6 +67,36 @@ TEST(SelectTileSizeTest, L1TileSizeTest) {
             (TileSize{32, 32, 64}));
 }
 
+TEST(SelectTileSizeTest, L2TileSizeTest) {
+  // The input params are {memoryLimit, numBytesA, numBytesB, numBytesC,
+  // numBytesAcc, bufferDepthA, bufferDepthB, bufferDepthC, bufferDepthAcc,
+  // inputM, inputN, inputK, vectorM, vectorN, vectorK}, maxL1TileM, maxL1TileN.
+  EXPECT_EQ(
+      (selectL2TileSizes(
+          {524288, 4, 4, 0, 4, 2, 2, 0, 2, 512, 512, 64, 4, 4, 8}, 32, 32)),
+      (TileSize{256, 128, 64}));
+  EXPECT_EQ(
+      (selectL2TileSizes({524288, 4, 4, 0, 4, 2, 2, 0, 2, 512, 32, 32, 4, 4, 8},
+                         32, 32)),
+      (TileSize{512, 32, 32}));
+  EXPECT_EQ(
+      (selectL2TileSizes(
+          {524288, 4, 4, 0, 4, 2, 2, 0, 2, 32, 128, 128, 4, 4, 8}, 32, 32)),
+      (TileSize{32, 128, 128}));
+  EXPECT_EQ(
+      (selectL2TileSizes(
+          {524288, 2, 2, 0, 4, 2, 2, 0, 2, 512, 512, 64, 4, 4, 8}, 32, 32)),
+      (TileSize{256, 128, 64}));
+  EXPECT_EQ(
+      (selectL2TileSizes(
+          {524288, 1, 1, 0, 4, 2, 2, 0, 2, 1024, 1024, 64, 4, 4, 8}, 32, 32)),
+      (TileSize{256, 128, 64}));
+  EXPECT_EQ((selectL2TileSizes({524288 * 8, 1, 1, 1, 2, 2, 2, 2, 0, 1024,
+                                4096 * 4, 512, 4, 4, 8},
+                               64, 64)),
+            (TileSize{1024, 1024, 512}));
```

Comment:
Could you add some comment to each test case to explain what case it is exactly trying to test and make sure we're covering most cases as well as add some edge cases? For example: are we testing non-divisible inputs, what if memory limit is `0` etc?

Reply:
You do check for it here: https://github.com/nod-ai/iree-amd-aie/pull/1231/files#diff-88fce979e5c13b4309da836589d63398f0d479e819ed81e18a251418b0f5132bR75 ?

Reply:
I added new tests for that.